android/jni/com/mapswithme/maps/editor/Editor.cpp

```diff
@@ -351,16 +362,41 @@ Java_com_mapswithme_maps_editor_Editor_nativeCreateMapObject(JNIEnv *, jclass, j
         ("Couldn't create mapobject, wrong coordinates of missing mwm"));
 }
 
+JNIEXPORT void JNICALL
+Java_com_mapswithme_maps_editor_Editor_nativeInitFeatureCategories(JNIEnv * env, jclass clazz)
+{
+  g_featureCategories = make_unique<osm::NewFeatureCategories>(g_framework->NativeFramework()->GetEditorCategories());
+}
```

Comment:
nativeInitFeatureCategories был всегда, он форсит как раз пересоздание объекта с нуля. При этом очищаются уже добавленные синонимы для языка, что может быть важно при переключении языка, например, чтобы поиск не работал по старому языку. Поэтому пока сделал так.

> On Apr 29, 2016, at 11:54 AM, Viktor Govako <notifications@github.com> wrote:
> 
> In android/jni/com/mapswithme/maps/editor/Editor.cpp <https://github.com/mpimenov/omim/pull/3#discussion_r61549675>:
> 
> > @@ -351,16 +362,41 @@ Java_com_mapswithme_maps_editor_Editor_nativeCreateMapObject(JNIEnv *, jclass, j
> >          ("Couldn't create mapobject, wrong coordinates of missing mwm"));
> >  }
> >  
> > +JNIEXPORT void JNICALL
> > +Java_com_mapswithme_maps_editor_Editor_nativeInitFeatureCategories(JNIEnv * env, jclass clazz)
> > +{
> > +  g_featureCategories = make_unique<osm::NewFeatureCategories>(g_framework->NativeFramework()->GetEditorCategories());
> > +}
> Если Framework::GetEditorCategories всегда возвращает корректный результат и он один и тот же :)
> то эта функция не нужна. И глобальная переменная g_featureCategories не нужна. Достаточно статическую в контексте функции GetFeatureCategories(). Вся инициализация пройдет один раз при первом вызове функции.
> 
> Если же важен момент инициализации (мало ли для Framework:: GetEditorCategories нужны какие-то условия), тогда GetFeatureCategories() может и лишняя.
> 
> Я написал самы первый коммент исходя из того текущего кода когда nativeInitFeatureCategories не было, а была копипаста в 2-х функциях.
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub <https://github.com/mpimenov/omim/pull/3/files/f3c136ffbac36134c474800acd05453d3c49b032#r61549675>